Stuff/harmonic_oscilator_obj_or/gaussian.cpp

```diff
@@ -32,16 +33,35 @@ double Gaussian::evaluateSecondDerivative(){
     double res = 0.0;
     for(int i=0; i<this->s->getNParticles(); i++){
         for(int j=0; j<this->s->getDimension(); j++){
-            res = res + pow(this->s->particles[i]->getPosition().at(j), 2);
+            res = res + pow(this->s->particles[i]->getPosition().at(j), 2); // particle i at its position 3 element vector alla dimensione j  
         }
     }
     res = res*4*pow(this->alpha, 2);
     for(int i=0; i<(this->s->getDimension()*this->s->getNParticles()); i++){
         res = res - 2*this->alpha;
     }
+    //std::cout << "res1 " << evaluateAll() << endl;
     return res*evaluateAll();
 }
 
+vector<double> Gaussian::evaluateGradient(){
+    vector<double> Grad;
+    Grad.push_back(1);
+    Grad.push_back(1);
+    Grad.push_back(1);
+    for(int i=0; i<this->s->getNParticles(); i++){ // for on all the particles
+        for(int j=0; j<this->s->getDimension(); j++){ // for on all the dimentions
+            Grad[j] = -2*this->alpha*this->s->particles[i]->getPosition().at(j);   
```

Comment:
Hmmm are you sure of this? Because for different i and same j you overwrite the previous value of the gradient with a new one, Did you mean to sum the new value to the previous one or I'm missing the purpose of this part of the code? In case you wanted to sum, rember to initialise at 0 not 1 @AlexsashaV4 @emistaff97